guide for aws lambda dotnet (#2670)
getsentry/sentry-docs: src/platforms/dotnet/guides/aws-lambda/index.mdx

```

You can combine this integration with a logging library like `log4net`, `NLog`, or `Serilog` to include both request data as well as your logs as breadcrumbs. The logging ingrations also capture events when an error is logged.

We also recommend updating your app/server to run on a more current version of the .NET Framework, such as .NET 4.7.2.

### Configuring

All `ASP.NET Core` configurations are valid here. But one configuration in particular is relevant.

`FlushOnCompletedRequest` ensures all events are flushed out. This is beacause the general ASP.NET Core hooks for when the process is exiting are not guaranteed to run in a serverless environment. This setting ensures that no event is lost if AWS recycles the process.

```csharp

public class LambdaEntryPoint : Amazon.Lambda.AspNetCoreServer.APIGatewayProxyFunction
{
    protected override void Init(IWebHostBuilder builder)
    {
        builder
            // Add Sentry
            .UseSentry(o =>
            {
              o.Dsn = "__PUBLIC_DSN__";
              // Required in Serverless environments
              o.FlushOnCompletedRequest = true;
            })
            .UseStartup<Startup>();
    }
}
```

> Check out the [Sentry ASP.NET Core](aspnetcore) documentation for the complete set of options.

Author: ```suggestion
Check out the [Sentry ASP.NET Core](/platforms/dotnet/guides/aspnetcore/) documentation for the complete set of options.
```

> Check out the [Sentry ASP.NET Core](/aspnetcore) documentation for the complete set of options.

Author: ```suggestion
Check out the [Sentry ASP.NET Core](aspnetcore) documentation for the complete set of options.
```

> Check out the [Sentry ASP.NET Core](/platforms/dotnet/guides/aspnetcore/) documentation for the complete set of options.

Author: ```suggestion
Check out the <PlatformLink to="/platforms/dotnet/guides/aspnetcore/">ASP.NET Core SDK</PlatformLink> documentation for the complete list of options.
```

> Check out the <PlatformLink to="/platforms/dotnet/guides/aspnetcore/">ASP.NET Core SDK</PlatformLink> documentation for the complete list of options.

Author: ```suggestion
Check out the [Sentry ASP.NET Core](/platforms/dotnet/guides/aspnetcore/) documentation for the complete set of options.
```

```

```shell {tabTitle:.NET Core CLI}
dotnet add package Sentry.AspNetCore -v {{ packages.version('sentry.dotnet.aspnetcore') }}
```

You can combine this integration with a logging library like `log4net`, `NLog` or `Serilog` to include both request data
as well as your logs as breadcrumbs. The logging ingrations also capture events when an error is logged.

On top of that, I advise you update your app/server to run on a more current version of the .NET Framework. Specifically, the latest, .NET 4.7.2.

### Configuring

All `ASP.NET Core` configurations are valid here. But one configuration in particular is relevant.

`FlushOnCompletedRequest` makes sure all events are flushed out. This is beacause the general ASP.NET Core hooks for when the process is exiting 
are not guaranteed to run in a serverless environment. This setting makes sure no event is lost in case AWS decides to recycle the process.

```csharp

public class LambdaEntryPoint : Amazon.Lambda.AspNetCoreServer.APIGatewayProxyFunction
{
    protected override void Init(IWebHostBuilder builder)
    {
        builder
            // Add Sentry
            .UseSentry(o =>
            {

>               o.Dsn = "__PUBLIC_DSN__;

Author: ```suggestion
              o.Dsn = "__PUBLIC_DSN__";
```

              // Required in Serverless environments
              o.FlushOnCompletedRequest = true;
            })
            .UseStartup<Startup>();
    }
}
```

Check out the [Sentry ASP.NET Core](/aspnetcore) documentation for the complete set of options.

## Samples

> - A [samples on GitHub](https://github.com/getsentry/sentry-dotnet/tree/main/samples/Sentry.Samples.Aws.Lambda.AspNetCoreServer) demonstrates Sentry on AWS Lambda. (**C#**)

Reviewer: ```suggestion
- Our [samples on GitHub](https://github.com/getsentry/sentry-dotnet/tree/main/samples/Sentry.Samples.Aws.Lambda.AspNetCoreServer) demonstrate Sentry on AWS Lambda. (**C#**)
```
